modules/user/user.service.js

```diff
@@ -0,0 +1,112 @@
+const { AuthenticationError, UserInputError } = require('apollo-server');
+const bcrypt = require('bcryptjs');
+const jwt = require('jsonwebtoken');
+const User = require('./user.model');
+const {
+  validateRegisterInput,
+  validateLoginInput,
+} = require('../../utils/validateUser');
+const generateToken = require('../../utils/createToken');
+const verifyUser = require('../../utils/verifyUser');
+
+class UserService {
+  getAllUsers() {
+    return User.find();
+  }
+
+  getUserById(id, data) {
+    const { token } = data.headers || data.cookies || '';
+    if (!token) {
+      throw new AuthenticationError(JSON.stringify({ message: 'no token' }));
+    }
+
+    if (verifyUser(token)) return User.findById(id);
+  }
+
+  updateUser(id, user) {
+    return User.findByIdAndUpdate(id, user);
```

Comment:
Please, add check that the user exists and throw `404` if the user is not found

Reply:
Also, please, add validation of data to update in user

Reply:
fixed